### PR TITLE
Bug #13890: DbTarget log transaction

### PR DESCRIPTION
**Pull request by shirase**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13890

### CODE REVIEW

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thank you!

Love the cloning solution.

File: framework/log/DbTarget.php [hunk 1]
```diff
@@ -60,6 +60,12 @@ public function init()
      */
     public function export()
     {
+        if ($this->db->getTransaction()) {
+            $this->db = clone $this->db;
+            $this->db->pdo = null;
```
Comment (cebe) [inline framework/log/DbTarget.php]: sorry for the late reply, but does this work in a master-slave setup? The situation is a bit more complex there: https://github.com/yiisoft/yii2/blob/1c07c429cf54e58ccb494f05ed889e6dce284bbe/framework/db/Connection.php#L578-L597

also transaction property of the DB is not reset so for each call to export another connection is opened even though there is no transaction on current connection anymore.

Comment (samdark) [inline framework/log/DbTarget.php]: master-slave should work well. I see no reason for it not to.

Transaction property — yes. Should be reset. Good catch.

Comment (cebe) [inline framework/log/DbTarget.php]: see #14121

Comment (samdark) [inline framework/log/DbTarget.php]: Ah. It could be slave transaction...

**Pull request merged**